### DISCUSSION TITLE
sam build vs deploy parameter overrides

### DISCUSSION THREAD

**Discussion by royassis** [aws/aws-sam-cli#5054] [Questions]
Hey there

I understand the logic of parameter during the sam build, but how are they used in sam deploy?

**Comment by super132** [comment]
Do you have any specific examples for us to understand what's your question?

**Reply by royassis** [reply]
Ok, 
I'm passing a CodeArtifact token to my SAM template, via a Parameter CodeArtifactAuthToken, which is only used during the build, this parameter doesn't have a default value. 
Whenever I'm deploying my template SAM cli asks me to enter some value to that same parameter.
